pkg/apis/provisioning/v1alpha5/kubelet_args.go

```diff
@@ -0,0 +1,22 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha5
+
+// KubeletArgs define args to be used when configuring kubelet on provisioned nodes
+type KubeletArgs struct {
+	// The IP address for in-cluster DNS resolution
+	//+optional
+	ClusterDNSIP string `json:"clusterDnsIp"`
```

Comment:
A couple approaches here

1. `type KubeletArgs []string`
2. `type KubeletArgs map[string]string`
3. `type KubeletArgs map[string]interface{}` <-- not sure if possible with crd generation
4. `type KubeletArgs struct { ...expandedovertime... }`
5.  Use this directly https://github.com/kubernetes/kubelet/blob/master/config/v1beta1/types.go#L88

I wonder how far we would get with option 1. It follows from ContainerSpec, which I like. It does mean that we have to do a bit of parsing for bottlerocket, but I'm not sure that this will be too hard.

I also like option 5, though it relies on regularly updating the kubelet dependency. It's the most future proof, but may expand our CRD significantly. We may be able to hack the CRD generation so that it's an "empty" to avoid too much bloat. It doesn't seem crazy to me that a node controller would be aware of a kubelet config though.

Reply:
I talked with @suket22 about this and we both felt it was important to keep this strongly typed. Otherwise, we're going to be doing lots of specific key checks (like this DNS cluster IP setting) because it doesn't go directly into the normal arg string. And because some require specific handling, we wanted to introduce only the ones we're currently supporting at the time (why we didn't want to go with option 5).

Reply:
If we're not going with option 5, can we at least go with an explicit subset of it? i.e., same key names, json annotations, etc. It should be as if we're using it, but we've disallowed most of the keys.

Reply:
And done 👍